include/spdlog/sinks/win_eventlog_sink.h

```diff
@@ -218,13 +220,23 @@ class win_eventlog_sink : public base_sink<Mutex>
     {
         using namespace internal;
 
+        bool succeeded;
         memory_buf_t formatted;
         base_sink<Mutex>::formatter_->format(msg, formatted);
         formatted.push_back('\0');
-        LPCSTR lp_str = static_cast<LPCSTR>(formatted.data());
 
-        auto succeeded = ::ReportEvent(event_log_handle(), eventlog::get_event_type(msg), eventlog::get_event_category(msg), event_id_,
-            current_user_sid_.as_sid(), 1, 0, &lp_str, nullptr);
+#ifdef SPDLOG_WCHAR_TO_UTF8_SUPPORT
+        auto buf = std::wstring_convert<std::codecvt_utf8<wchar_t>>().from_bytes(formatted.data());
```

Comment:
```std::wstring_convert<std::codecvt_utf8>``` is deprecated. Please use  ```details::os::wstr_to_utf8buf(..)``` instead. See usage example [here](https://github.com/gabime/spdlog/blob/9ce39a470fa7adc8b101fdd5e4325e6c371f675c/include/spdlog/logger.h#L243).